### PR TITLE
Jh.kj.create route for child account

### PR DESCRIPTION
**Pull request by Tvcz**
Worked with @chromium-52 on a route for creating a child account for an admin parent account in the `PARENT_ACCOUNTS` table.

Route accessible via: `POST api/v1/protected/user/create_child`

[ClickUp Ticket](https://app.clickup.com/t/27pm0r6)

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java [hunk 1]
```diff
@@ -211,4 +230,27 @@ public void changePrivilegeLevel(
       throw new WrongPasswordException();
     }
   }
+
+  @Override
+  public void createChildUser(JWTData userData, NewChildRequest newChildRequest) {
+    isAdminCheck(userData.getPrivilegeLevel());
+
+    db.transaction(configuration -> {
```
Comment (chromium-52) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java:238]: @rymaju Is this what you meant by wrapping it with a transaction?

Comment (Tvcz) [reply service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java:238]: Sure! - ryan

File: service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java [hunk 2]
```diff
@@ -63,6 +71,17 @@ private UsersRecord userRecordExistsCheck(UsersRecord user, UserDoesNotExistExce
     return user;
   }
 
+  /**
+   * Throws an exception if the user is not an admin or super admin.
+   *
+   * @param level the privilege level of the user calling the route
+   */
+  private void isAdminCheck(PrivilegeLevel level) {
```
[the remark this answers is not in the document]
Comment (chromium-52) [reply service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java:79]: Got it! I'll do this in a separate PR.

**Review by SofieDunt — APPROVED** [review]
💯 Looks good to me!

**Review by chromium-52 — APPROVED** [review]
LGTM!

**Pull request merged**